src/managed/OpenLiveWriter.PostEditor/Tables/PixelPercent.cs

```diff
@@ -0,0 +1,143 @@
+using System;
+using System.Diagnostics;
+using System.Globalization;
+
+namespace OpenLiveWriter.PostEditor.Tables
+{
+    public enum PixelPercentUnits
+    {
+        Undefined,
+        Pixels,
+        Percentage
+    }
+
+    /// <summary>
+    /// Represents a value that can be a number of pixels or a percentage
+    /// </summary>
+    [DebuggerDisplay("{Value} {Units}")]
+    public struct PixelPercent
+    {
+        public PixelPercent(int value, PixelPercentUnits units) : this()
+        {
+            Units = units;
+
+            if (value < 0)
+                throw new ArgumentOutOfRangeException("value", value, "value must be greater than zero");
+
+            if (value > 100 && units == PixelPercentUnits.Percentage)
+                throw new ArgumentOutOfRangeException("value", value,
+                    "value when percent must be less or equal to 100");
+
+            Value = value;
+        }
+
+        public PixelPercent(string text, IFormatProvider provider, PixelPercentUnits units)
+        {
+            if (string.IsNullOrEmpty(text))
+            {
+                Value = 0;
+                Units = PixelPercentUnits.Undefined;
+            }
+            else
+            {
+                var s = text.Trim();
+                Units = units;
+
+                var value = int.Parse(s, provider);
```

Comment:
Would it be better to use TryParse here instead of regular Parse, would get rid of the need of waiting for an exception to be thrown in the CanParse method.